src/Microsoft.AspNet.Mvc.Razor/Razor/PreCompileViews/RazorPreCompiler.cs

```diff
@@ -58,93 +64,126 @@ public virtual void CompileViews([NotNull] IBeforeCompileContext context)
             }
         }
 
-        protected virtual IReadOnlyList<RazorFileInfo> CreateCompilationDescriptors(
-                                                            [NotNull] IBeforeCompileContext context)
+        protected virtual IEnumerable<RazorFileInfo> CreateCompilationDescriptors(
+            [NotNull] IBeforeCompileContext context)
         {
-            var list = new List<RazorFileInfo>();
+            var filesToProcess = new List<RelativeFileInfo>();
+            GetFileInfosRecursive(root: string.Empty, razorFiles: filesToProcess);
 
-            foreach (var info in GetFileInfosRecursive(string.Empty))
-            {
-                var descriptor = ParseView(info, context);
+            var razorFiles = new RazorFileInfo[filesToProcess.Count];
+            var syntaxTrees = new SyntaxTree[filesToProcess.Count];
+            var parallelOptions = new ParallelOptions { MaxDegreeOfParallelism = MaxDegreesOfParallelism };
```

Comment:
here I agree w/ @pranavkm.  mucking with the `TaskScheduler` or cancellation token seems far below the fold compared to exposing `MaxDegreesOfParallelism`